app/controllers/job_invocations_controller.rb

```diff
@@ -72,6 +72,26 @@ def preview_hosts
     render :partial => 'job_invocations/preview_hosts_list'
   end
 
+  def cancel
+    @job_invocation = resource_base.find(params[:id])
+    result = @job_invocation.cancel(params[:force])
+
+    if result
+      flash[:notice] = if params[:force]
+                         _('Trying to abort the job')
+                       else
+                         _('Trying to cancel the job')
+                       end
+    else
+      flash[:warning] = if params[:force]
+                          _('The job cannot be aborted at the moment.')
+                        else
+                          _('The job cannot be cancelled at the moment.')
+                        end
+    end
+    redirect_to :back
```

Comment:
This doesn't work in rails 5, see https://github.com/theforeman/foreman-tasks/pull/308/files on how to transition that